### PR TITLE
Post ghc-9.2.1 config cleanup

### PR DESCRIPTION
**Pull request by jneira**
* I've removed grain coarsed allow-newer's and uncomment all subpackages to track what precise dependencies we need to get full ghc-9.2.1 support
* Some changes in .cabal has being unveiled

<a href="https://gitpod.io/#https://github.com/haskell/haskell-language-server/pull/2582"></a>

### CODE REVIEW

**Comment by jneira** [comment]
floskell has no allow-newer in cabal-ghc921.project so it seems to me it has more chances to work

File: haskell-language-server.cabal [hunk 1]
```diff
@@ -283,7 +283,7 @@ common qualifyImportedNames
 -- formatters
 
 common floskell
-  if flag(floskell) && (impl(ghc < 9.2.1) || flag(ignore-plugins-ghc-bounds))
+  if flag(floskell) || flag(ignore-plugins-ghc-bounds))
```
Comment (michaelpj) [inline haskell-language-server.cabal:286]: trailing paren

Comment (jneira) [reply haskell-language-server.cabal]: ugh I did the same error I just corrected in the other pr  🤦‍♂️

File: cabal-ghc921.project [hunk 2]
```diff
@@ -6,33 +6,32 @@ packages:
          ./ghcide
          ./hls-plugin-api
          ./hls-test-utils
+<<<<<<< HEAD
```
Comment (michaelpj) [inline cabal-ghc921.project:9]: mismerge

**Review by michaelpj — APPROVED** [review]

**Pull request merged**